backends/ceed-backend-list.h

```diff
@@ -5,24 +5,24 @@
 // listed, and also to define weak symbol aliases for backends that are not
 // configured.
 
-MACRO(CeedRegister_Avx_Blocked)
-MACRO(CeedRegister_Avx_Serial)
-MACRO(CeedRegister_Cuda)
-MACRO(CeedRegister_Cuda_Gen)
-MACRO(CeedRegister_Cuda_Shared)
-MACRO(CeedRegister_Hip)
-MACRO(CeedRegister_Hip_Gen)
-MACRO(CeedRegister_Hip_Shared)
-MACRO(CeedRegister_Magma)
-MACRO(CeedRegister_Magma_Det)
-MACRO(CeedRegister_Memcheck_Blocked)
-MACRO(CeedRegister_Memcheck_Serial)
-MACRO(CeedRegister_Occa)
-MACRO(CeedRegister_Opt_Blocked)
-MACRO(CeedRegister_Opt_Serial)
-MACRO(CeedRegister_Ref)
-MACRO(CeedRegister_Ref_Blocked)
-MACRO(CeedRegister_Tmpl)
-MACRO(CeedRegister_Tmpl_Sub)
-MACRO(CeedRegister_Xsmm_Blocked)
-MACRO(CeedRegister_Xsmm_Serial)
+MACRO(CeedRegister_Avx_Blocked, 1, "/cpu/self/avx/blocked")
+MACRO(CeedRegister_Avx_Serial, 1, "/cpu/self/avx/serial")
+MACRO(CeedRegister_Cuda, 1, "/gpu/cuda/ref")
+MACRO(CeedRegister_Cuda_Gen, 1, "/gpu/cuda/gen")
+MACRO(CeedRegister_Cuda_Shared, 1, "/gpu/cuda/shared")
+MACRO(CeedRegister_Hip, 1, "/gpu/hip/ref")
+MACRO(CeedRegister_Hip_Gen, 1, "/gpu/hip/gen")
+MACRO(CeedRegister_Hip_Shared, 1, "/gpu/hip/shared")
+MACRO(CeedRegister_Magma, 2, "/gpu/cuda/magma", "/gpu/hip/magma")
+MACRO(CeedRegister_Magma_Det, 2, "/gpu/cuda/magma/det", "/gpu/hip/magma/det")
+MACRO(CeedRegister_Memcheck_Blocked, 1, "/cpu/self/memcheck/blocked")
+MACRO(CeedRegister_Memcheck_Serial, 1, "/cpu/self/memcheck/serial")
+MACRO(CeedRegister_Occa, 4, "/cpu/self/occa", "/cpu/openmp/occa", "/gpu/hip/occa", "/gpu/cuda/occa")
+MACRO(CeedRegister_Opt_Blocked, 1, "/cpu/self/opt/blocked")
+MACRO(CeedRegister_Opt_Serial, 1, "/cpu/self/opt/serial")
+MACRO(CeedRegister_Ref, 1, "/cpu/self/ref/serial")
+MACRO(CeedRegister_Ref_Blocked, 1, "/cpu/self/ref/blocked")
+MACRO(CeedRegister_Tmpl, 1, "/cpu/self/tmpl")
+MACRO(CeedRegister_Tmpl_Sub, 1, "/cpu/self/tmpl/sub")
+MACRO(CeedRegister_Xsmm_Blocked, 1, "/cpu/self/xsmm/blocked")
+MACRO(CeedRegister_Xsmm_Serial, 1, "/cpu/self/xsmm/serial")
```

Comment:
This is a pretty clean solution, though I really don't love the redundancy and don't know how this can extend to programmatically generated resource strings (I think we've discussed this for multiple devices and maybe other contexts). One of my thoughts was that `ceed-backend-list.h` could some day be automatically generated by the build system, from scanning the source directories for matching declarations.

When we load plugins, there is definitely no centralized list. That would put them on unequal footing from in-package backends. I would have said that a simpler, and more generic error message would be sufficient, but now that you've put this together, I'm not going to object. I would like to hear other opinions.

Reply:
I agree that the redundancy felt bad while I was building this. It would have been easier/cleaner if we didn't have a backend that responded to substantially different names (OCCA).

I'm definitely ok omitting this code if we prefer, I mostly put it together to see 1) if I could somewhat cleanly and 2) if we like it.

Reply:
I think the feature it brings is nice, I'm not too concerned about the style, but I might not see the issue you foresee.